### ISSUE TITLE
Local Archive Option

### ISSUE THREAD

**Issue by sherlockedBBC** [Kovah/LinkAce#496]
In the spirit of selfhosting i would like to archive or backup links to a local archive instead of archive.org. 

Idealy it would be an integration of [SingleFile](https://github.com/gildas-lormeau/SingleFile) into LinkAce to automaticly download Websites as a single HTML file. A bonus would be the automatic download of media files like youtube-videos via something like [jdownloader](https://jdownloader.org/).

_Alternatives:_
[Archivebox](https://archivebox.io/) exists but the options to download doesn´t work to well in my short experience and LinkAce has the superior interface. [Linkwarden](https://github.com/Daniel31x13/link-warden) looks promising but is not as refined as LinkAce.

**Comment by Kovah** [comment]
I will definitely add an option to choose ArchiveBox for archiving links. However, this is only possible when ArchiveBox ships with a working REST API:
- https://github.com/ArchiveBox/ArchiveBox/issues/496

**Comment by piegamesde** [comment]
ArchiveBox can import from RSS feeds. Maybe LinkAce could export an RSS endpoint? Maybe also per list or per tag, so that you could have an `archivebox` tag instead of archiving everything

**Comment by Kovah** [comment]
RSS feeds already exist for anything you need, either without authentication containing private links, or without authentication for everything that is also accessible to guests.

See the docs for more details: https://www.linkace.org/docs/v1/application/rss-feeds/

**Comment by nezred** [comment]
> Idealy it would be an integration of [SingleFile](https://github.com/gildas-lormeau/SingleFile) into LinkAce to automaticly download Websites as a single HTML file. A bonus would be the automatic download of media files like youtube-videos via something like [jdownloader](https://jdownloader.org/).

I agree, although I do think having the option to just manually import attachments would be amazing, since I doubt it would be as difficult, and It can be useful for any edge cases, such as saving the executable for some free indie game that bookmark points to, or more niche video hosting sites aside from youtube.

I had been considering moving my stuff to LinkAce, because it looked good, and the only thing it was missing to have everything I need was the ability to upload attachments to my bookmarks (I know automatic singlefile website download is not exactly attachments, but idc)

**Comment by Kovah** [comment]
Short update on this:
- Archivebox still has no API in a stable release.
- Archive.is still has no API.

I consider researching for alternative solutions (like [browsertrix](https://github.com/webrecorder/browsertrix)) that might enable people to properly archive their web pages.
